src/Type.h

```diff
@@ -0,0 +1,378 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "proto/substrait/algebra.pb.h"
+#include "proto/substrait/type.pb.h"
```

Comment:
It's not a good idea to expose protobuf headers in this library's headers, IMHO.

Reply:
do you have any suggestions?

Reply:
The usual solution is to not expose any protobuf-generated types in your own headers. So define your own C++ classes and enums for any types that might have an equivalent in the protobuf definitions...

This is the solution we use in Parquet (with Thrift), Arrow (with flatbuffers), Arrow Flight (with protobuf).

(I'll add that even with that, protobuf still has issues in non-trivial scenarios: https://github.com/substrait-io/substrait/issues/249 ; but it will have less of them)

Reply:
Sounds good,  I will refactor to use our own type enums instead of protobuf based type enums.

Reply:
updated